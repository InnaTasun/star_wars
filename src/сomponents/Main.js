import mainLogo from '../images/mainLogo.jpeg';
import FilmsNav from './FilmsNav';
import styled from 'styled-components';
import { useFetch } from '../hooks/useFetch';
import * as URLS from '../constants/urls';

function Main({ className }) {
  const movies = useFetch(URLS.FILMS_URL);

  return (
    <div className={className}>
      <div className='main-logo'>
        <img className='main-logo--img' src={mainLogo} alt='main logo' />
      </div>
      <h3>MOVIES</h3>
      <FilmsNav dataArr={movies} />
      <br />
    </div>
  );
}

const StyledMain = styled(Main)`
  .main-logo {
    height: 400px;
    background-color: black;
    text-align: center;
    margin-bottom: 40px;
    box-shadow: 7px 7px 7px rgba(0, 0, 0, 0.5);
    &--img {
      height: 100%;
    }
  }
`;

export default StyledMain;
