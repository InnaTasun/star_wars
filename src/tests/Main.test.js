import React from 'react';
import { render, screen } from '@testing-library/react';
import Main from '../сomponents/Main';

describe('Main component', () => {
  test('renders', () => {
    render(<Main />);
    expect(screen.getByText(/movies/i)).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
  });

  test('renders with or without className', () => {
    let container = render(<Main className='main-class' />).container;
    let firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('main-class');

    container = render(<Main className='test-class' />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('test-class');

    container = render(<Main />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).not.toContain('main-class');
  });
});

test("Main snapshot", () => {
  const { container } = render(<Main />);
  expect(container).toMatchSnapshot();
});