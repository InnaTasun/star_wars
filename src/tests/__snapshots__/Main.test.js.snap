// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN iYKAki"
  >
    <div
      class="main-logo"
    >
      <img
        alt="main logo"
        class="main-logo--img"
        src="mainLogo.jpeg"
      />
    </div>
    <h3>
      MOVIES
    </h3>
    <ul
      class="sc-gsDKAQ enRQhp films-list"
    />
    <br />
  </div>
</div>
`;
